solr/core/src/test/org/apache/solr/spelling/DirectSolrSpellCheckerTest.java

```diff
@@ -88,6 +88,45 @@ public void testOnlyMorePopularWithExtendedResults() throws Exception {
         "//lst[@name='spellcheck']/lst[@name='suggestions']/lst[@name='fox']/arr[@name='suggestion']/lst/int[@name='freq']=2",
         "//lst[@name='spellcheck']/bool[@name='correctlySpelled']='true'"
     );
-  }  
+  }
+
+  @Test
+  public void testMaxQueryLength() throws Exception {
+    testMaxQueryLength(true);
+    testMaxQueryLength(false);
+  }
+
+  private void testMaxQueryLength(Boolean limitQueryLength) throws Exception {
+
+    DirectSolrSpellChecker checker = new DirectSolrSpellChecker();
+    NamedList spellchecker = new NamedList();
```

Comment:
We should use generics here: NamedList<Object> spellchecker = new NamedList<>();

Reply:
I've just used what you suggested here - am not too familiar with how this works.